src/ShopBundle/pkgExtranet/objects/db/TShopDataExtranetUser.class.php

```diff
@@ -442,7 +445,11 @@ public function AddArticleIdToNoticeList($iArticleId, $iAmount = 1)
         } else {
             $oNoticeListItem = TdbShopUserNoticeList::GetNewInstance();
             /** @var $oNoticeListItem TdbShopUserNoticeList */
-            $aData = array('shop_article_id' => $iArticleId, 'date_added' => date('Y-m-d H:i:s'), 'data_extranet_user_id' => $this->id, 'amount' => $iAmount);
+            $aData = [
+                'shop_article_id' => $iArticleId,
+                'date_added' => date('Y-m-d H:i:s'),
+                'amount' => $iAmount,
```

Comment:
We do have a permanent notice list in the database. This contains the extranet user id. When the user logs in, the notice list in the database is merged with the one from the cookie. I assume this will break if we remove the id from the cookie data. See for example in `\TShopDataExtranetUser::PostLoginHookMergeTemporaryNoticeListWithDatabaseHistory`

Reply:
You are right in this particular case. But the point stands. It is a BC break for the benefit of maybe staying under an arbitrary cookie limit. I don't think that's worth it to be honest.

Reply:
IMHO it is not a BC break if it fixes something that is broken already if you add +25 items to your notice list. Yes, we don't know the exact amount you can add, because it depends on the products you add, their name, description etc., but 25 is a limit which works. It is not a perfect solution, but it fixes the current behavior, which is really a problem that costs conversion, because if someone adds more than ~25 elements you can't use the shop anymore, till you drop your cookies.

Reply:
So your suggestion is a limit of 17? That is what would fit with this additional field.

Reply:
Yes, that would be a plausible solution

Reply:
ok.